api/lib/config/setting.js

```diff
@@ -41,28 +39,40 @@ module.exports = function(settingsObj) {
 
   }
   settings.validate = function(){
+    if(typeof(settings.port) == "undefined" || settings.port == null){
+      return{valid:false, message:"The port is required"}
+    }
     if(typeof(settings.port) != "number") {
       return {valid:false,message:"The port must be a number"};
     }
     if (settings.port < 0 || settings.port > 65536) {
       return {valid:false,message:"The value of port must between 0 and 65536"};
     }
 
-    if(typeof(settings.db.maxConnections) != "number") {
+    if(typeof(settings.db.maxConnections) == "undefined" || settings.db.maxConnections == null){
       return {valid:false,message:"maxConnections is required"};
     }
+    if(typeof(settings.db.maxConnections) != "number") {
+      return {valid:false,message:"maxConnections must be a number"};
+    }
     if(settings.db.maxConnections < 0) {
       return {valid:false,message:"maxConnections must be greater than 0"};
     }
-    if(typeof(settings.db.minConnections) != "number") {
+    if(typeof(settings.db.minConnections) == "undefined" || settings.db.minConnections == null){
       return {valid:false,message:"minConnections is required"};
     }
+    if(typeof(settings.db.minConnections) != "number") {
+      return {valid:false,message:"minConnections must be a number"};
+    }
     if(settings.db.minConnections < 0) {
       return {valid:false,message:"minConnections must be greater than 0"};
     }
-    if(typeof(settings.db.idleTimeout) != "number") {
+    if(!settings.db.idleTimeout || settings.db.idleTimeout == null){
```

Comment:
Little different from the way done for the rest with typeof?
 Also can make a generic function to do the required check  and another one to check if number?